packages/framework-core/src/services/graphql/graphql-query-generator.ts

```diff
@@ -141,75 +131,118 @@ export class GraphQLQueryGenerator {
-        return
-      }
+      const primitiveType = this.typeInformer.getPrimitiveExtendedType(prop.typeInfo.type)
+
       args[prop.name] = {
-        type: this.generateFilterFor(prop.type),
+        type: primitiveType === Array ? this.generateArrayFilterFor(prop) : this.generateFilterFor(primitiveType),
       }
     })
     return args
   }
 
-  private canFilter(graphQLType: GraphQLNonInputType): boolean {
-    return graphQLType instanceof GraphQLScalarType && graphQLType != GraphQLJSONObject
-  }
+  private generateArrayFilterFor(property: PropertyMetadata): GraphQLInputObjectType {
+    const filterName = `${property.name}PropertyFilter`
 
-  private generateFilterFor(type: AnyClass): GraphQLInputObjectType {
-    const filterName = `${type.name}PropertyFilter`
     if (!this.generatedFiltersByTypeName[filterName]) {
-      const graphQLValueType = this.typeInformer.getGraphQLTypeFor(type)
+      const propFilters: GraphQLInputFieldConfigMap = {}
+      property.typeInfo.parameters.forEach(() => {
+        propFilters.includes = {
+          ...propFilters.includes,
+          type: GraphQLJSONObject,
+        }
+      })
+
       this.generatedFiltersByTypeName[filterName] = new GraphQLInputObjectType({
         name: filterName,
-        fields: {
-          operation: { type: new GraphQLNonNull(this.operationEnumFor(type)) },
-          values: { type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(graphQLValueType))) },
-        },
+        fields: propFilters,
       })
     }
     return this.generatedFiltersByTypeName[filterName]
   }
 
-  private operationEnumFor(type: AnyClass): GraphQLEnumType {
-    const operationEnumName = `${type.name}Operations`
-    if (!this.generatedOperationEnumsByTypeName[operationEnumName]) {
-      this.generatedOperationEnumsByTypeName[operationEnumName] = new GraphQLEnumType({
-        name: operationEnumName,
-        values: this.generateOperationEnumValuesFor(type),
-      })
+  private generateFilterFor(type: AnyClass): GraphQLInputObjectType {
+    const filterName = `${type.name}PropertyFilter`
+    if (!this.generatedFiltersByTypeName[filterName]) {
+      const primitiveType = this.typeInformer.getPrimitiveExtendedType(type)
+      const graphQLPropType = this.typeInformer.getGraphQLTypeFor(primitiveType)
+      let fields: Thunk<GraphQLInputFieldConfigMap> = {}
+
+      if (!this.typeInformer.canFilter(graphQLPropType)) {
+        const properties = getPropertiesMetadata(type)
+        this.typeInformer.generateGraphQLTypeFromMetadata({ class: type, properties })
+
+        let nestedProperties: GraphQLInputFieldConfigMap = {}
+        for (const prop of properties) {
+          const propPrimitiveType = this.typeInformer.getPrimitiveExtendedType(prop.typeInfo.type)
+          const property = { [prop.name]: { type: this.generateFilterFor(propPrimitiveType) } }
+          nestedProperties = { ...nestedProperties, ...property }
```

Comment:
I'm a bit confused about the use of ellipsis here. Would this code be similar to:
```suggestion
          const propPrimitiveType = this.typeInformer.getPrimitiveExtendedType(prop.typeInfo.type)
          nestedProperties[prop.name] = { type: this.generateFilterFor(propPrimitiveType) }
```